src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -54,122 +65,186 @@ const getMessageForTask = (isCurrentUser: boolean, taskType: TaskType) => {
-                })}
-                data-testid="task-card"
-            >
-                <Comment
-                    avatarRenderer={avatar => (
-                        <div className="bcs-task-avatar">
-                            {avatar}
-                            <TaskTypeIcon width={20} height={20} className="bcs-task-avatar-badge" />
-                        </div>
-                    )}
-                    created_at={created_at}
-                    created_by={created_by.target}
-                    currentUser={currentUser}
-                    id={id}
-                    isPending={isPending}
-                    onDelete={onDelete}
-                    onEdit={onEdit}
-                    permissions={taskPermissions}
-                    tagged_message={description}
-                    translatedTaggedMessage={translatedTaggedMessage}
-                    translations={translations}
-                    type={COMMENT_TYPE_TASK}
-                    getAvatarUrl={getAvatarUrl}
-                    getUserProfileUrl={getUserProfileUrl}
-                    mentionSelectorContacts={mentionSelectorContacts}
-                    getMentionWithQuery={getMentionWithQuery}
-                    userHeadlineRenderer={userLinkInstance => {
-                        return (
-                            <FormattedMessage
-                                {...getMessageForTask(!!currentUserAssignment, task_type)}
-                                values={{ user: userLinkInstance }}
+    handleCreateSuccess = () => {
+        this.setState({ isEditing: false });
+    };
+
+    handleCreateError = (e: ElementsXhrError) => {
+        this.setState({ modalError: e });
+    };
+
+    getUsersFromTask = (): SelectorItems => {
+        const { assigned_to } = this.props;
+
+        return (
+            assigned_to &&
+            assigned_to.entries.map(taskCollab => {
+                const retVal = {
+                    ...taskCollab.target,
+                    item: {},
+                    value: taskCollab.target.id,
+                    text: taskCollab.target.name,
+                };
+
+                return retVal;
+            })
+        );
+    };
+
+    render() {
+        const {
+            assigned_to,
+            created_at,
+            created_by,
+            currentUser,
+            due_at,
+            error,
+            features,
+            getAvatarUrl,
+            getMentionWithQuery,
+            getUserProfileUrl,
+            id,
+            isPending,
+            mentionSelectorContacts,
+            description,
+            onAssignmentUpdate = noop,
+            onDelete,
+            onEdit,
+            permissions,
+            status,
+            task_type,
+            translatedTaggedMessage,
+            translations,
+        } = this.props;
+
+        const { modalError, isEditing } = this.state;
+
+        const taskPermissions = {
```

Comment:
The API does; new tasks API response lack `can_edit`, and the Comment component doesn't use `can_update`. Since the existing code reuses `Comment`, looks like the initial version of this check maps the value so the comment component knows what to do